### PR TITLE
EPMDJ-1877: Update agent settings page

### PR DESCRIPTION
**Pull request by RomanAchilov**
Issue: EPMDJ-1877

### CODE REVIEW

File: src/pages/agent-settings-page/system-setting-form/unlocking-system-settings-form-modal/unlocking-system-settings-form-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+interface Props {
+  className?: string;
+  isOpen: boolean;
+  onToggle: (value: boolean) => void;
+  setUnlocked: (value: boolean) => void;
+}
+
+const unlockingSystemSettingsFormModal = BEM(styles);
+
+export const UnlockingSystemSettingsFormModal = unlockingSystemSettingsFormModal(
```
Comment (pavyarov) [inline src/pages/agent-settings-page/system-setting-form/unlocking-system-settings-form-modal/unlocking-system-settings-form-modal.tsx]: dunno, i think we need generec modal for that case, somethink like <ConfirmActionModal />

File: src/pages/agent-settings-page/agent-settings-page.tsx [hunk 2]
```diff
@@ -1,64 +1,100 @@
...
-
-        {selectedTab === 'general' ? (
-          <GeneralSettingsForm agent={agent} showMessage={showMessage} />
-        ) : (
-          <SystemSettingsForm agent={agent} showMessage={showMessage} />
-        )}
+        {tabsComponents
+          .filter(({ name }) => name === selectedTab)
+          .map(({ name, component }) => (
+            <div key={name}>{component}</div>
```
Comment (pavyarov) [inline src/pages/agent-settings-page/agent-settings-page.tsx]: and why u need <div> ?

File: src/pages/agent-settings-page/agent-settings-page.tsx [hunk 3]
```diff
@@ -72,5 +108,16 @@ export const AgentSettingsPage = withRouter(
 );
 
 const HeaderIcon = agentSettingsPage.headerIcon(Icons.Settings);
-const UnregisterAgentButton = agentSettingsPage.unregisterAgentButton(Button);
+const ToggleAgent = agentSettingsPage.toggleAgent(AgentStatusToggler);
```
Comment (pavyarov) [inline src/pages/agent-settings-page/agent-settings-page.tsx]: AgentStatus maybe?

File: src/pages/agent-settings-page/agent-settings-page.tsx [hunk 4]
```diff
@@ -72,5 +108,16 @@ export const AgentSettingsPage = withRouter(
 );
 
 const HeaderIcon = agentSettingsPage.headerIcon(Icons.Settings);
-const UnregisterAgentButton = agentSettingsPage.unregisterAgentButton(Button);
+const ToggleAgent = agentSettingsPage.toggleAgent(AgentStatusToggler);
+const ErrorMessage = agentSettingsPage.errorMessage(Panel);
+const ErrorMessageIcon = agentSettingsPage.errorMessageIcon(Icons.Warning);
 const Tabs = agentSettingsPage.tabsPanel(TabsPanel);
+
+async function toggleStandby(agentId: string, showError: (message: string) => void) {
```
Comment (pavyarov) [inline src/pages/agent-settings-page/agent-settings-page.tsx]: move to api

File: src/pages/agent-settings-page/plugins-settings/add-plugins-modal/add-plugins-modal.tsx [hunk 5]
```diff
@@ -1,88 +1,88 @@
...
-  setIsAddPluginOpen(false);
-};
+import * as React from 'react';
+import { BEM } from '@redneckz/react-bem-helper';
+import axios from 'axios';
+
+import { Modal } from '../../../../components';
+import { Button } from '../../../../forms';
+import { SelectableList } from './selectable-list';
+import { useWsConnection } from '../../../../hooks';
```
Comment (pavyarov) [inline src/pages/agent-settings-page/plugins-settings/add-plugins-modal/add-plugins-modal.tsx]: move to line 7

File: src/pages/agent-settings-page/plugins-settings/add-plugins-modal/add-plugins-modal.tsx [hunk 6]
```diff
@@ -1,88 +1,88 @@
...
+import { Plugin } from '../../../../types/plugin';
+import { defaultAdminSocket } from '../../../../common/connection';
```
Comment (pavyarov) [inline src/pages/agent-settings-page/plugins-settings/add-plugins-modal/add-plugins-modal.tsx]: move to line 8

File: src/pages/agent-settings-page/plugins-settings/add-plugins-modal/add-plugins-modal.tsx [hunk 7]
```diff
@@ -1,88 +1,88 @@
...
+interface Props {
+  className?: string;
+  isOpen: boolean;
+  onToggle: (arg: boolean) => void;
+  agentId: string;
+}
+
+const addPluginModal = BEM(styles);
+
+export const AddPluginsModal = addPluginModal(({ className, isOpen, onToggle, agentId }: Props) => {
```
Comment (pavyarov) [inline src/pages/agent-settings-page/plugins-settings/add-plugins-modal/add-plugins-modal.tsx]: i think we should rename this component, AddPluginSidebar

File: src/pages/agent-settings-page/plugins-settings/add-plugins-modal/add-plugins-modal.tsx [hunk 8]
```diff
@@ -1,88 +1,88 @@
...
+});
+
+const Header = addPluginModal.header('div');
+const Content = addPluginModal.content('div');
+const PluginsList = addPluginModal.pluginsList('div');
+const Actions = addPluginModal.actions('div');
+const AddPluginButton = addPluginModal.addPluginButton(Button);
+const CandelButton = addPluginModal.cancelButton(Button);
+
+const loadPlugins = ({
```
Comment (pavyarov) [inline src/pages/agent-settings-page/plugins-settings/add-plugins-modal/add-plugins-modal.tsx]: refactor that

File: src/pages/agent-settings-page/plugins-settings/add-plugins-modal/selectable-list/selectable-list.tsx [hunk 9]
```diff
@@ -1,46 +1,46 @@
...
+import * as React from 'react';
+import { BEM } from '@redneckz/react-bem-helper';
+
+import { Inputs } from '../../../../../forms';
+
+import styles from './selectable-list.module.scss';
+
+interface Props {
+  className?: string;
+  data: any[];
```
Comment (pavyarov) [inline src/pages/agent-settings-page/plugins-settings/add-plugins-modal/selectable-list/selectable-list.tsx]: we can calculate type for data

File: src/pages/agent-settings-page/plugins-settings/get-selected-plugins-actions.tsx [hunk 10]
```diff
@@ -1,60 +1,60 @@
...
-};
-
-// TODO: write unit test
-export const getSelectedPluginsMap = (plugins: any[] = [], selectedPluginsIds: string[]) =>
-  plugins.reduce<{ [key: string]: boolean }>(
-    (acc, { id = '', status }) =>
-      selectedPluginsIds.includes(id) ? { ...acc, [id]: status === AGENT_STATUS.ONLINE } : acc,
-    {},
-  );
+import axios from 'axios';
```
Comment (pavyarov) [inline src/pages/agent-settings-page/plugins-settings/get-selected-plugins-actions.tsx]: remove this file

File: src/pages/agent-settings-page/system-setting-form/system-settings-form.module.scss [hunk 11]
```diff
@@ -33,26 +33,55 @@
...
   }
 
-  &__instructions {
+  &__icon-lock-wrapper {
+    display: flex;
     color: $slate-gray;
+    margin-left: 8px;
+    cursor: pointer;
+
+    &--unlocked {
```
Comment (pavyarov) [inline src/pages/agent-settings-page/system-setting-form/system-settings-form.module.scss]: icon-lock-wrapper has modifier unlocked rly?

File: src/pages/agent-settings-page/system-setting-form/system-settings-form.tsx [hunk 12]
```diff
@@ -64,26 +66,67 @@ export const SystemSettingsForm = systemSettingsForm(
...
-                        index !== value.length - 1 ? acc + item + '\n' : acc + item,
-                      '',
-                    )
-                  }
-                  placeholder="Package name 1&#10;Package name 2&#10;Package name 3&#10;and so on."
-                />
+                <FieldName>
+                  Project Package(s)
+                  <IconLockWrapper
+                    unlocked={unlocked && !isUnlockingModalOpened}
```
Comment (pavyarov) [inline src/pages/agent-settings-page/system-setting-form/system-settings-form.tsx]: !isUnlockingModalOpened ?

File: src/pages/agent-settings-page/system-setting-form/system-settings-form.tsx [hunk 13]
```diff
@@ -64,26 +66,67 @@ export const SystemSettingsForm = systemSettingsForm(
...
+                    onClick={() => {
+                      unlocked ? setUnlocked(false) : setIsUnlockingModalOpened(true);
+                    }}
+                  >
+                    {unlocked ? (
+                      <Icons.Info />
+                    ) : (
+                      <Tooltip
+                        message={
+                          <SecuredMessage direction="column">
```
Comment (pavyarov) [inline src/pages/agent-settings-page/system-setting-form/system-settings-form.tsx]: u can use <div> without panel no?

File: src/pages/agent-settings-page/system-setting-form/system-settings-form.tsx [hunk 14]
```diff
@@ -64,26 +66,67 @@ export const SystemSettingsForm = systemSettingsForm(
...
+                            <span>Secured from editing.</span>
+                            <span> Click to unlock.</span>
+                          </SecuredMessage>
+                        }
+                        position="top"
```
Comment (pavyarov) [inline src/pages/agent-settings-page/system-setting-form/system-settings-form.tsx]: default position is 'top' don't need to specify that

File: src/pages/agent-settings-page/system-setting-form/system-settings-form.tsx [hunk 15]
```diff
@@ -64,26 +66,67 @@ export const SystemSettingsForm = systemSettingsForm(
...
+                      </Tooltip>
+                    )}
+                  </IconLockWrapper>
+                </FieldName>
+                <Panel verticalAlign="start">
+                  <Field
+                    name="packagesPrefixes"
+                    component={ProjectPackages}
+                    parse={(value) => value.split(/\n/)}
+                    format={(value) =>
```
Comment (pavyarov) [inline src/pages/agent-settings-page/system-setting-form/system-settings-form.tsx]: should be moved to formattors with unit test

File: src/pages/agent-settings-page/agent-settings-page.tsx (hunk 2 above)
Comment (RomanAchilov) [reply src/pages/agent-settings-page/agent-settings-page.tsx]: Each child in a list should have a unique "key" prop.

**Pull request merged**